### DISCUSSION TITLE
I want to get the Cartwright number or Doodson Number of each  tide constituent. Can this software be used to achieve this?

### DISCUSSION THREAD

**Discussion by q1165600785** [pyTMD/pyTMD#263] [General]
I want to input   an constituent  then get it Cartwright number or Doodson Number

**Comment by tsutterley** [comment]
Not currently but on my to-do list.

**Comment by tsutterley** [comment]
Added in PR #270 and part of [v2.1.0](https://github.com/tsutterley/pyTMD/releases/tag/2.1.0)
```
>>> import pyTMD
>>> pyTMD.arguments.doodson_number('m2')
255.555
>>> pyTMD.arguments.doodson_number('m2', formalism='Cartwright')
array([2., 0., 0., 0., 0., 0.])
>>>  pyTMD.arguments.doodson_number(['m2','s2'])
{'m2': 255.555, 's2': 273.555}
>>>  pyTMD.arguments.doodson_number(['m2','s2'], formalism='Cartwright')
{'m2': array([2., 0., 0., 0., 0., 0.]), 's2': array([ 2.,  2., -2.,  0.,  0.,  0.])}
```

**Reply by q1165600785** [reply]
First of all, I would like to express my gratitude to the professor for his implementation and hard work. As I advance my research, I need to obtain the calculation of the astronomical initial phase. Based on the astronomical parameters and the Doodson Number, I have read the Python code for the implementation of utide, which is similar to the matalb s-tide or Python version. It cannot be separated and relies on external files. I would like to obtain a pure Python version of the implementation that can be ported. Does the professor have this plan

**Reply by tsutterley** [reply]
I'm not planning on replicating `utide` functionality. There is some basic solve capability for decomposing a time series into the amplitude and phase of individual constituents. I doubt it meets the sophistication of those more mature packages. Is there something specific you are looking to do?

**Reply by q1165600785** [reply]
I am researching the issue of correcting the height difference of ocean tides in geometric leveling. According to national standards, it requires obtaining the angular frequency and initial phase of each tidal component. I want to use Pytmd to complete these calculations. Teacher, you have plans to establish a table about angular frequency and calculate the initial phase. Teacher, you have already achieved the initial phase by multiplying six astronomical coefficients and numbers, but it still depends on a phase offest from cont. semi. I feel that this implementation is difficult for me
